geode-junit/src/main/java/org/apache/geode/test/junit/rules/ExecutorServiceRule.java

```diff
@@ -236,6 +246,98 @@ public void execute(Runnable command) {
     return CompletableFuture.supplyAsync(supplier, executor);
   }
 
+  /**
+   * Returns the {@code Thread}s that are directly in the {@code ExecutorService}'s
+   * {@code ThreadGroup} excluding subgroups.
+   */
+  public Set<Thread> getThreads() {
+    return threadFactory.getThreads();
+  }
+
+  /**
+   * Returns an array of {@code Thread Ids} that are directly in the {@code ExecutorService}'s
+   * {@code ThreadGroup} excluding subgroups. {@code long[]} is returned to facilitate using JDK
+   * APIs such as {@code ThreadMXBean#getThreadInfo(long[], int)}.
+   */
+  public long[] getThreadIds() {
+    Set<Thread> threads = getThreads();
+    long[] threadIds = new long[threads.size()];
+
+    int i = 0;
+    for (Thread thread : threads) {
+      threadIds[i++] = thread.getId();
+    }
+
+    return threadIds;
+  }
+
+  /**
+   * Returns formatted call stacks of the {@code Thread}s that are directly in the
+   * {@code ExecutorService}'s {@code ThreadGroup} excluding subgroups.
+   */
+  public String dumpThreads() {
```

Comment:
There are other things you could include in the thread dumps, like threadInfo.getLockInfo() and threadInfo.getLockedMonitors(), that could help folks.  See OSProcess.formatThreadInfo().

Reply:
Yeah, I wish I could make OSProcess.formatThreadInfo() public and call that from geode-junit. Need to think about how to best do this.